src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -34,19 +34,19 @@ public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, b
             if (outputBuffer == null)
                 ThrowHelper.ThrowArgumentNull(ThrowHelper.ExceptionArgument.outputBuffer);
 
-            // For now, only convert 3 bytes to 4
-            Span<byte> input = inputBuffer.AsSpan(inputOffset, InputBlockSize);
-            Span<byte> output = outputBuffer.AsSpan(outputOffset, OutputBlockSize);
+            int inputBlocks = inputCount / InputBlockSize;
+            int outputBlocks = (outputBuffer.Length - outputOffset) / OutputBlockSize;
+            int blocksToTransform = Math.Min(inputBlocks, outputBlocks);
+            Debug.Assert(blocksToTransform >= 1);
```

Comment:
This assert can be hit if somebody passes _inputCount = 0_ or _outputCount = int.MaxValue_.

Reply:
The method throws earlier if inputCount < InputBlockSize.

And outputBuffer.Length can't be int.MaxValue, as that's larger than the max possible array size.

Reply:
outputOffset = outputBuffer.Length (or Length - {1, 2, 3}) would reduce blocksToTransform to 0.

Previously that would have thrown from the AsSpan.

Reply:
> Previously that would have thrown from the AsSpan.

I'll add explicit validation for that upfront then.  The error coming from AsSpan wouldn't have been particularly informative.